src/Workspaces/Core/Portable/Remote/RemoteHostClient.cs

```diff
@@ -78,6 +78,91 @@ public static string CreateClientId(string prefix)
             return $"VS ({prefix}) ({Guid.NewGuid().ToString()})";
         }
 
+        public static Task<RemoteHostClient?> TryGetClientAsync(Project project, CancellationToken cancellationToken)
+        {
+            if (!RemoteSupportedLanguages.IsSupported(project.Language))
+            {
+                return SpecializedTasks.Default<RemoteHostClient?>();
+            }
+
+            return TryGetClientAsync(project.Solution.Workspace, cancellationToken);
+        }
+
+        public static Task<RemoteHostClient?> TryGetClientAsync(Workspace workspace, CancellationToken cancellationToken)
+        {
+            var service = workspace.Services.GetService<IRemoteHostClientService>();
+            if (service == null)
+            {
+                return SpecializedTasks.Default<RemoteHostClient?>();
+            }
+
+            return service.TryGetRemoteHostClientAsync(cancellationToken);
+        }
+
+        /// <summary>
+        /// Creates <see cref="SessionWithSolution"/> for the <paramref name="serviceName"/> if possible, otherwise returns <see langword="null"/>.
+        /// </summary>
+        public async Task<SessionWithSolution?> TryCreateSessionAsync(string serviceName, Solution solution, CancellationToken cancellationToken, object? callbackTarget = null)
```

Comment:
📝 Saving an overload here isn't worth the reordering it forces. It would be much better to just have both of the following:

```
public Task<SessionWithSolution?> TryCreateSessionAsync(string serviceName, Solution solution, CancellationToken cancellationToken)
public Task<SessionWithSolution?> TryCreateSessionAsync(string serviceName, Solution solution, object? callbackTarget, CancellationToken cancellationToken)
```

Reply:
It's saving 6 overloads.

Reply:
IDE generally doesn't have an issue with overloads.  If anything, i'd be ok with making *none* of these variables optional, and just telling features to pass in 'null' here if htey want no callback from the OOP process.

Reply:
OK, i'll make it non-optional.